.github/workflows/actions.yml

```diff
@@ -0,0 +1,50 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Padl-Extensions-CI/CD
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ macos-latest, ubuntu-latest, windows-latest ]
+        python-version: ["3.7", "3.8", "3.9"]
+        exclude:
+          - os: windows-latest
+            python-version: 3.9
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e ".[all]"
+        pip install -r requirements-test.txt
+    - name: Setup Java 11 for torchserve
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'zulu'
+        java-version: '11'
+    - name: Test with pytest
+      run: |
+        python -m pytest -s --cov=padl --cov-report=xml
+    - name: "Upload coverage to Codecov"
+      if: ${{ matrix.python-version == '3.8' && matrix.os == 'ubuntu-latest' }}
+      uses: codecov/codecov-action@v2
+      with:
+        fail_ci_if_error: true
```

Comment:
@wuhu I think it would be a good idea to add `padl-extensions` to our codecov account to be able to save code coverage reports like for `padl`. What do you think?